modules/react/src/deckgl.js

```diff
@@ -89,25 +87,18 @@ export default class DeckGL extends React.PureComponent {
   }
 
   // Callbacks
-
-  // Forward callback and then call forceUpdate to guarantee that sub components update
-  _onResize(params) {
-    this.setState(params);
-    this.props.onResize(params);
+  _redrawDeck() {
+    this.deck._drawLayers('React update');
   }
 
-  // Forward callback and then call forceUpdate to guarantee that sub components update
-  _onViewStateChange(params) {
-    // Let app know that view state is changing, and give it a chance to change it
-    const viewState = this.props.onViewStateChange(params) || params.viewState;
+  _customRender() {
+    const viewports = this.deck.viewManager.getViewports();
 
-    // If initialViewState was set on creation, auto track position
-    if (this.state.viewState) {
-      this.setState({
-        viewState: Object.assign({}, this.state.viewState, {
-          [params.viewId]: viewState
-        })
-      });
+    if (viewports !== this.viewports) {
+      // Viewports have changed, update children props first
```

Comment:
what is the reason of this comment `Viewports have changed, update children props first`?